router/src/main/java/io/micronaut/web/router/DefaultRouteBuilder.java

```diff
@@ -428,6 +428,15 @@ private UriRoute buildBeanRoute(HttpMethod httpMethod, String uri, BeanDefinitio
             targetMethod.getValue(Produces.class, MediaType[].class).ifPresent(produces ->
                     this.producesMediaTypes = Arrays.asList(produces)
             );
+
+            this.conditions.add(req -> {
+                List<MediaType> consumes = this.acceptedMediaTypes;
+                if (consumes != null && !consumes.isEmpty()) {
+                    return req.getContentType().map(consumes::contains).orElse(true);
```

Comment:
Might need a `|| consumes.contains(MediaType.ALL)`

Also JAXRS and Spring both support patterns like `application/**` or `**/xml`, although I had not planned to implement that until 1.1.x

Reply:
I've pushed the change to support `MediaType.ALL`